libs/content/workspace/src/task_manager.rs

```diff
@@ -304,9 +429,31 @@ impl TaskManager {
 
         let sync_to_launch =
             if should_sync { mem::take(&mut tasks.queued_syncs).into_iter().next() } else { None };
-
-        let any_to_launch =
-            !loads_to_launch.is_empty() || !saves_to_launch.is_empty() || sync_to_launch.is_some();
+        let sync_status_update_to_launch = if should_update_sync_status {
+            mem::take(&mut tasks.queued_sync_status_updates)
+                .into_iter()
+                .next()
+        } else {
+            None
+        };
+
+        let any_to_launch = !loads_to_launch.is_empty()
+            || !saves_to_launch.is_empty()
+            || sync_to_launch.is_some()
+            || sync_status_update_to_launch.is_some();
+        if any_to_launch {
+            debug!(
+                "launching {} loads, {} saves, {} syncs, and {} sync status updates; {} loads, {} saves, {} syncs, and {} sync status updates remain queued",
+                loads_to_launch.len(),
+                saves_to_launch.len(),
+                sync_to_launch.is_some() as usize,
+                sync_status_update_to_launch.is_some() as usize,
+                tasks.queued_loads.len(),
+                tasks.queued_saves.len(),
+                tasks.queued_syncs.len(),
+                tasks.queued_sync_status_updates.len()
+            );
```

Comment:
for your consideration, you may find the following syntax more ergonomic / readable: https://docs.rs/tracing/latest/tracing/macro.debug.html#examples